test/tests/amqp/test_amqp_poller_alert.py

```diff
@@ -0,0 +1,367 @@
+                "typeId field not correct!  expect {0}, get {1}"
+                .format(expected_payload['typeId'], amqp_body_json['typeId']))
+            self.assertEquals(
+                amqp_body_json['action'], expected_payload['action'],
+                "action field not correct!  expect {0}, get {1}"
+                .format(expected_payload['action'], amqp_body_json['action']))
+            self.assertEquals(
+                amqp_body_json['severity'],
+                expected_payload['severity'],
+                "serverity field not correct!" .format(
+                    expected_payload['severity'],
+                    amqp_body_json['severity']))
+            self.assertNotEquals(
+                amqp_body_json['createdAt'],
+                {},
+                "createdAt field is empty!")
+            self.assertNotEquals(
+                amqp_body_json['data'],
+                {},
+                "data field is empty!")
+        except ValueError as e:
+            self.fail(
+                "FAILURE - expected key is missing in the AMQP message!{0}".format(e))
+
+    def test_sel_alert(self):
+        # The SEL log poller test will inject a CPU IERR event in the node BMC and moniter the AMQP message queue
+        node_collection = test_api_utils.get_node_list_by_type("compute")
+        nodeid = ""
+        # Check is skupack is installed. If no skupack installed, we won't get any SEL alert
+        skupack_installed = self.check_skupack()
+        for dummy in node_collection:
+            nodeid = node_collection[
+                random.randint(
+                    0, len(node_collection) - 1)]
+            if fit_common.rackhdapi(
+                '/api/2.0/nodes/' +
+                    nodeid)['json']['name'] != "Management Server":
+                break
+        logs.debug_2('Checking OBM setting...')
+        node_obm = fit_common.rackhdapi(
+            '/api/2.0/nodes/' + nodeid)['json']['obms']
+        if node_obm == []:
+            self.assertTrue(
+                self._apply_obmsetting_to_node(nodeid),
+                "Fail to apply obm setting!")
+        # Clear sel to avoid the sel log full
+        test_api_utils.run_ipmi_command_to_node(nodeid, "sel clear")
+        # Find the SEL poller id. This id will be used to validate AMQP message body.
+        pollerid = self._get_ipmi_poller_by_node(nodeid, "selEntries")
+        logs.debug('launch AMQP thread for sel alert')
+        sel_worker = AmqpWorker(
+            exchange_name="on.events",
+            topic_routing_key="polleralert.sel.#." + nodeid,
+            external_callback=self.amqp_callback,
+            timeout=200)
+        sel_worker.setDaemon(True)
+        sel_worker.start()
+        # Send out sel iERR injection.
+        command = "raw 0x0a 0x44 0x01 0x00 0x02 0xab 0xcd 0xef 0x00 0x01 0x00 0x04 0x07 0x02 0xef 0x00 0x00 0x00"
+        test_api_utils.run_ipmi_command_to_node(nodeid, command)
+        if skupack_installed:

```

Comment:
This check "if skupack_installed" should be done after line 265.  ie. Fail the test if skupack_installed is false before doing any of the work to set things up.
Otherwise, what I read, is you set up all the requirements,  start the ampq worker, inject an error, but then will exit before you actually verify anything "if skupack_installed is false".  If it is true, then you actually verify the test, but if false, you just exit and don't raise an error.  So if false, your test will always pass if all the setup went ok.  :-)